src/main/java/uk/gov/ons/ctp/common/rest/RestClient.java

```diff
@@ -187,24 +220,32 @@ public RestTemplate getRestTemplate() {
               + response.getStatusCode();
       log.error(errorMessage);
       throw new ResponseStatusException(
-          mapToExternalStatus(response.getStatusCode()), "Internal processing error");
+          mapToExternalStatus(response.getStatusCode()), "Internal processing error. No response.");
     }
-    try {
-      responseObject = objectMapper.readValue(responseBody, clazz);
-    } catch (IOException e) {
-      log.error(
-          "Failed to convert response to DTO object. Path: '"
-              + uriComponents
-              + "' Status: "
-              + response.getStatusCodeValue()
-              + " ResponseBody: '"
-              + responseBody
-              + "'",
-          e);
-      throw new ResponseStatusException(
-          HttpStatus.INTERNAL_SERVER_ERROR, "Internal processing error");
+    // Convert response body to target object
+    if (clazz == String.class) {
+      responseObject = (T) responseBody;
+    } else {
+      // Convert Json to target type
+      try {
+        responseObject = objectMapper.readValue(responseBody, clazz);
```

Comment:
As discussed please remove all of this code to do with objectMapper - it is not needed

Reply:
Removed